ci-operator/step-registry/gcp/bastion/pre/gcp-bastion-pre-commands.sh

```diff
@@ -0,0 +1,278 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM
+
+if [ X"${CREATE_BASTION}" == X"no" ]; then
+  echo "CREATE_BASTION is set to 'no', so nothing to do." && exit 0
+fi
+
+function generate_proxy_ignition() {
+cat > /tmp/proxy.ign << EOF
+{
+  "ignition": {
+    "config": {},
+    "security": {
+      "tls": {}
+    },
+    "timeouts": {},
+    "version": "3.0.0"
+  },
+  "passwd": {
+    "users": [
+      {
+        "name": "core",
+        "sshAuthorizedKeys": [
+          "${ssh_pub_key}"
+        ]
+      }
+    ]
+  },
+  "storage": {
+    "files": [
+      {
+        "path": "/etc/squid/passwords",
+        "contents": {
+          "source": "data:text/plain;base64,${HTPASSWD_CONTENTS}"
+        },
+        "mode": 420
+      },
+      {
+        "path": "/etc/squid/squid.conf",
+        "contents": {
+          "source": "data:text/plain;base64,${SQUID_CONFIG}"
+        },
+        "mode": 420
+      },
+      {
+        "path": "/etc/squid.sh",
+        "contents": {
+          "source": "data:text/plain;base64,${SQUID_SH}"
+        },
+        "mode": 420
+      },
+      {
+        "path": "/etc/squid/proxy.sh",
+        "contents": {
+          "source": "data:text/plain;base64,${PROXY_SH}"
+        },
+        "mode": 420
+      }
+    ]
+  },
+  "systemd": {
+    "units": [
+      {
+        "contents": "[Unit]\nWants=network-online.target\nAfter=network-online.target\n[Service]\n\nStandardOutput=journal+console\nExecStart=bash /etc/squid.sh\n\n[Install]\nRequiredBy=multi-user.target\n",
+        "enabled": true,
+        "name": "squid.service"
+      },
+      {
+        "dropins": [
+          {
+            "contents": "[Service]\nExecStart=\nExecStart=/usr/lib/systemd/systemd-journal-gatewayd \\\n  --key=/opt/openshift/tls/journal-gatewayd.key \\\n  --cert=/opt/openshift/tls/journal-gatewayd.crt \\\n  --trust=/opt/openshift/tls/root-ca.crt\n",
+            "name": "certs.conf"
+          }
+        ],
+        "name": "systemd-journal-gatewayd.service"
+      }
+    ]
+  }
+}
+EOF
+}
+
+# TODO: move to image
+curl -L https://github.com/mikefarah/yq/releases/download/3.3.0/yq_linux_amd64 -o /tmp/yq && chmod +x /tmp/yq
+
+CONFIG="${SHARED_DIR}/install-config.yaml"
+
+PROXY_IMAGE=registry.ci.openshift.org/origin/4.5:egress-http-proxy
```

Comment:
Not: export "4.5" as RELEASE env var, so that we could update it without changing the scrip